packages/dds/task-manager/src/taskManager.ts

```diff
@@ -157,7 +157,7 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         super(id, runtime, attributes);
 
         this.opWatcher.on("volunteer", (taskId: string, clientId: string, local: boolean, messageId: number) => {
-            if (local) {
+            if (runtime.connected && local) {
```

Comment:
Maybe add a comment here for future reference, something like
```suggestion
            // We are tracking local ops from this connection.
            // Filter out local ops during "connecting" state since these were sent on the prior connection and were already cleared from the latestPendingOps.
            if (runtime.connected && local) {
```

Reply:
I've added the suggested comment.

I've also been looking into the test you mentioned, but it won't hit this case. As far as I understand it (I'm not too familiar with the node event handling loop), I think we're hitting this case in the load tests because of the [fault injection logic](https://github.com/microsoft/FluidFramework/blob/562a0722eec42a9c31eb2d256dc9f1ea0be720e9/packages/test/test-service-load/src/runner.ts#L246) that is calling setTimeout and emitting a disconnect event. I'm not sure it would be possible to write a unit test that would simulate this behavior without it being flaky, but I'm open to suggestions if I'm thinking about this incorrectly.

Reply:
Thanks - is this the missing test capability that we need to simulate?
1. Client sends a volunteer op prior to disconnect
2. Orderer sequences the volunteer op
3. Client disconnects (without seeing the volunteer ack)
4. Orderer sequences the client leave op
5. Client reconnects, and sees the volunteer ack

I agree I don't see an obvious way to split the send from seeing the ack with our current test infrastructure -- maybe log an issue to enhance our test infrastructure to support this, as I expect it would be useful for a variety of our DDSs.

Reply:
That's the correct capability we need to simulate. I've opened #10013 to address this missing functionality in our testing infrastructure.